MergerCli/utils/BatchStatusManager.cs

```diff
@@ -74,13 +152,42 @@ public bool IsBaseLayerNew()
 
         public string? GetLayerBatchIdentifier(string layer)
         {
-            if (this.States.ContainsKey(layer))
+            lock (_locker)
+            {
+                if (this.States.ContainsKey(layer))
+                {
+                    return this.States[layer].BatchIdentifier;
+                }
+                return null;
+            }
+        }
+
+        public long? GetTotalCompletedTiles(string layer)
+        {
+            lock (_locker)
             {
-                return this.States[layer].BatchIdentifier;
+                if (this.States.ContainsKey(layer))
```

Comment:
Instead of this whole function i think you can write `return this.States[layer]?.TotalCompletedTiles;`.